source/adapters/cuda/command_buffer.cpp

```diff
@@ -9,48 +9,154 @@
 //===----------------------------------------------------------------------===//
 
 #include "command_buffer.hpp"
+
 #include "common.hpp"
+#include "enqueue.hpp"
+#include "event.hpp"
+#include "kernel.hpp"
+#include "memory.hpp"
+#include "queue.hpp"
+
+ur_exp_command_buffer_handle_t_::ur_exp_command_buffer_handle_t_(
+    ur_context_handle_t hContext, ur_device_handle_t hDevice)
+    : Context(hContext),
+      Device(hDevice), CudaGraph{nullptr}, CudaGraphExec{nullptr}, RefCount{1} {
+  urContextRetain(hContext);
+  urDeviceRetain(hDevice);
+}
+
+// The ur_exp_command_buffer_handle_t_ destructor release all the memory objects
+// allocated for command_buffer managment
+ur_exp_command_buffer_handle_t_::~ur_exp_command_buffer_handle_t_() {
+  // Release the memory allocated to the Context stored in the command_buffer
+  urContextRelease(Context);
+
+  // Release the device
+  urDeviceRelease(Device);
+
+  // Release the memory allocated to the CudaGraph
+  cuGraphDestroy(CudaGraph);
+
+  // Release the memory allocated to the CudaGraphExec
+  cuGraphExecDestroy(CudaGraphExec);
+}
 
-/// Stub implementations of UR experimental feature command-buffers
+/// Helper function for finding the Cuda Nodes associated with the
```

Comment:
I am running `clang-format` on the files but it seems to be enforcing the couple of lines that are slightly over 80 columns, which seems odd.

Reply:
Since the CI will fail if the formatting isn't correct I think I'll need to leave this as is for now, since `clang-format` is considering this to be right.